tests/Unit/Evolution/Systems/CurvedScalarWave/Test_Characteristics.cpp

```diff
@@ -229,34 +233,71 @@ SPECTRE_TEST_CASE("Unit.Evolution.Systems.CurvedScalarWave.Characteristics",
   test_characteristics_compute_tags<1>();
   test_characteristics_compute_tags<2>();
   test_characteristics_compute_tags<3>();
+}
+
+namespace {
+template <size_t SpatialDim>
+void check_max_char_speed(const DataVector& used_for_size) noexcept {
+  CAPTURE(SpatialDim);
+  MAKE_GENERATOR(gen);
 
-  CHECK(CurvedScalarWave::ComputeLargestCharacteristicSpeed<1>::apply(
-            {{DataVector{1., -4., 3.4, 2., 5.},
-              DataVector{22., -8., 190., 6., 4.},
-              DataVector{1., -7., 31., 2., 5.},
-              DataVector{7., 8.9, 4., 2., 1.}}}) == 190.);
-  CHECK(CurvedScalarWave::ComputeLargestCharacteristicSpeed<1>::apply(
-            {{DataVector{1., 4., 3., 2., 5.}, DataVector{2., 8., -10., 6., 4.},
-              DataVector{1., 7., 3., -11., 5.},
-              DataVector{7., 3., 4., 2., 1.}}}) == 11.);
+  // Fraction of times the test can randomly fail
+  const double failure_tolerance = 1e-10;
+  // Minimum fraction of the claimed result that must be found in some
+  // random trial
+  const double check_minimum = 0.9;
+  // Minimum fraction of the claimed result that can be found in any
+  // random trial (generally only important for 1D where the random
+  // vector will point along the maximum speed direction)
+  const double check_maximum = 1.0 + 1e-12;
 
-  CHECK(CurvedScalarWave::ComputeLargestCharacteristicSpeed<2>::apply(
-            {{DataVector{1., -4., 3.4, 2., 5.},
-              DataVector{22., -8., 190., 6., 4.},
-              DataVector{1., -7., 31., 2., 5.},
-              DataVector{7., 8.9, 4., 2., 1.}}}) == 190.);
-  CHECK(CurvedScalarWave::ComputeLargestCharacteristicSpeed<2>::apply(
-            {{DataVector{1., 4., 3., 2., 5.}, DataVector{2., 8., -10., 6., 4.},
-              DataVector{1., 7., 3., -11., 5.},
-              DataVector{7., 3., 4., 2., 1.}}}) == 11.);
+  const double trial_failure_rate =
+      SpatialDim == 1
+          ? 0.1  // correct value is 0.0, but cannot take log of that
+          : (SpatialDim == 2 ? 1.0 - 2.0 * acos(check_minimum) / M_PI
+                             : check_minimum);
+
+  const size_t num_trials =
+      static_cast<size_t>(log(failure_tolerance) / log(trial_failure_rate)) + 1;
+
+  const auto lapse = TestHelpers::gr::random_lapse(&gen, used_for_size);
+  const auto shift =
+      TestHelpers::gr::random_shift<SpatialDim>(&gen, used_for_size);
+  const auto spatial_metric =
+      TestHelpers::gr::random_spatial_metric<SpatialDim>(&gen, used_for_size);
+  std::uniform_real_distribution<> gamma_1_dist(-5.0, 5.0);
```

Comment:
isn't gamma1 restricted to being positive for hyperbolicity?

Reply:
yup, but for the purpose of testing the code this shouldn't matter. I've changed the lower limit to zero though.